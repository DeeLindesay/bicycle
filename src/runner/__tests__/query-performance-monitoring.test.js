// @flow

import {runQuery, enablePerformanceMonitoring} from '../';
import loadSchema from '../../load-schema';

test('runQuery', async () => {
  enablePerformanceMonitoring();
  const log = [];
  const oldLog = console.log;
  (console: any).log = str => log.push(str);
  await runQuery(
    loadSchema({
      objects: [
        {
          name: 'Root',
          fields: {
            foo: {
              type: 'string',
              resolve() {
                return new Promise(resolve => setTimeout(() => resolve('foo'), 1000));
              },
            },
          },
        },
      ],
    }),
    {foo: true},
    {},
  );
  expect(log.join('\n').replace(/[0-9]+[a-z][a-z]?/g, 'TIME')).toMatchSnapshot();
  (console: any).log = oldLog;
});
