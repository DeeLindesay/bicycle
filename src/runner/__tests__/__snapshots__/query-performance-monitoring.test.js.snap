// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runQuery 1`] = `
"Query completed in TIME

Fields that took over TIME to resolve:

 * Root.foo - TIME (1 call)

Note that this is the **total** time to reolve the fields,
so you may be seeing some fields that are requested only
once, but are very slow, and some fields that are quick,
but are requested thousands of times."
`;
